subiquity/common/serialize.py

```diff
@@ -127,7 +140,8 @@ def _serialize_datetime(self, annotation, value, metadata, path):
     def _serialize_field(self, field, value, path):
         path = f'{path}.{field.name}'
         return {
-            field.name: self.serialize(field.type, value, field.metadata, path)
+            _field_name(field): self.serialize(
```

Comment:
When should _field_name be used, and when field.name?

Reply:
Well oops. You ask this question and instead of simply explaining, I rewrote everything a bit.

Basically you need to use _field_name() when looking at the serialized form, and field.name when looking at the rich objects. About half the usages only matter for the "path" that is inconsistently included in some error messages, so I rewrote everything to make the path tracking and error messages much more consistent. Sorry about that -- I do think the code is better for it though. I added some very simple tests of the error handling.